### ISSUE TITLE
[BUG/CRASH] Packet overload upon boss attacks and weapon abilities [EXCEPTION_ACCESS_VIOLATION (0xc0000005)]

### ISSUE THREAD

**Issue by NoRestForWicked** [mariumbacchus/Soulslike-Weaponry#162]
Example:
Minecraft: 1.20.1
Fabric: 0.15.11
Soulslike-Weaponry: 1.1.4.2

This is the crash I have been having issues with. It is the same issue the darkin scythe had before the most recent patch update (lovely patch btw). Upon the use of any attack that has particles, the client will have a packet overload and immediately crash. this happens inconstantly but many examples is The Nightfall's smash attack, the darkin blade's smash attack, comet spear calculated fall landing, Genesis fracture ability, Returning night and fallen icon smash attacks, all of the deity attacks, some night shade attacks, and I think there are more. but it regards every attack using particles and it's completely random when it happens. (By the way, you DID fix the darkin scythe's issue. So everything should be under the same list, but it seems like a lot)

No logs are generated as the game just reverts to before the crash for single player, and in multiplayer the only thing logged is the join and leaving of the player that crashed.

To reproduce, use any ability that uses a bunch of particles or fight a boss a bunch of times until the crash occurs.

Expected behavior: It shouldn't crash :)
_(Hey, if you're interested in joining an smp, you're more than welcome to join mine which is the main reason for me asking for the patches :D. Just thought I should mention lol)_

A video of me and another member of the smp fighting a supremely config-buffed fallen icon and then crashing:
https://youtu.be/c-3Qz3eLeiw

I have not tested it by itself on this version yet, but on version 1.1.3 these crashes were still prevalent while the mod was isolated.

**Comment by mariumbacchus** [comment]
Thanks for the video, seems like it still is an issue then. I will look into it, fixing the Umbral Trespass one did give some insight into why this happens.

And thanks for the invite! Maybe I'll join another time in the future, got a lot going on right now, but I'll let you know if the time comes.

**Comment by NoRestForWicked** [comment]
Yeah, it is still prevelant lol.

And of course, by all means. 

By the way, I will open a new thread for this but I've completed a theme (Ima have you judge it) for the returning knight/fallen icon. I feel like fallen icon should have a 2nd phase theme and since fallen icon and returning knight are essentially the same, they get the same theme.

**Comment by alltheArdy** [comment]
Hi! Was wondering if this was fixed in a recent update? A server we host has this mod in its modpack with the version 1.1.4.2, and we have been experiencing clientside crashes on certain boss attacks

Should I get the pack devs to update the mod somehow? (the version it is on is the latest according to curseforge)

**Comment by NoRestForWicked** [comment]
It is not fixed yet, but I do believe marium is doing as best they can to resolve the issue 🔥

**Comment by NoRestForWicked** [comment]
@mariumbacchus if by any chance could you give us an update on this situation? :D 
I have been having the issue of this happening all the time against fallen icon and all the other bosses. The bosses make the crash happen alot more than the weapons, but it happens with those too.

**Comment by mariumbacchus** [comment]
I am currently working through all the issues, but updating through 6 versions (fabric 1.18.2 to 1.20.4 + forge 1.18.2 and 1.20.1) takes time. I just read something interesting in the other crash logs about packets causing memory crash, I belive you read it yourself a few days ago too, namely forcing the Java version to 17.0.11 fixes the issues. If you could try this and see if that helps, it would be greatly appreciated.

**Comment by NoRestForWicked** [comment]
I appreciate you working on it, I thank you greatly. But If I may ask how would I force java version to 17.0.11 on fabric

**Comment by NoRestForWicked** [comment]
I am on java 17 when the crashes happen is there a difference between 17.011 and 17?

**Comment by mariumbacchus** [comment]
There might be, I don't really know, but I still want to test this out.
Referring to [this](https://github.com/mariumbacchus/Soulslike-Weaponry/issues/170#issuecomment-2307928875) post, apparently he installed Prism Launcher and managed to force the version there. As for how, it is a simple google search away: 
![image](https://github.com/user-attachments/assets/df854803-0473-41cc-b411-c95129c49dff)

**Comment by NoRestForWicked** [comment]
Huh, i will try out prism launcher and see if the crash still happens, and if it fixes Id like to find a way to do it with curseforge launcher

**Comment by NoRestForWicked** [comment]
YES, this is a fix! I used prism and installed java 17.0.11 and no boss or weapon could crash me. I had 5 of my testers help me battle fallen icon and the deities simultaneously, and they crashed a ton, and I did not. To be sure, I joined using my curseforge client and crashed equally as much as they did. However I rejoined with prism and did not crash once. I even spawned 20 night prowlers and everyone except me crashed immediately. Now I just need to test and see if I can change this on curseforge.

**Comment by NoRestForWicked** [comment]
Do note that java 17 alone does NOT work

**Comment by NoRestForWicked** [comment]
I figured out how to do it with curseforge, and it works as it does with prism. The only issue is that it is a tad annoying to do, as you have to manually download java 17.0.11 from https://download.oracle.com/java/17/archive/jdk-17.0.11_windows-x64_bin.exe
and then go to your minecraft settings in curseforge, and change the directory of java 17

**Comment by mariumbacchus** [comment]
Ok nice, now I at least know why this crash is happening. Thanks for testing it out, I'll try to figure out a fix.

**Comment by NoRestForWicked** [comment]
One of my admins made a tutorial for it, It is below.

Click the link to download Java 17.0.11: https://download.oracle.com/java/17/archive/jdk-17.0.11_windows-x64_bin.exe

Click the link to view the video: https://youtu.be/-JV00P-hdr0

**Comment by mariumbacchus** [comment]
Alright nice. Maybe I'll put this up on the mod page or in the README file. It is a bit much to do though, so I will still look for a potential fix.

**Comment by Andryusplay** [comment]
Is there a way to fix this on Tlauncher?

**Comment by mariumbacchus** [comment]
The fix is the same for every launcher, just make the correct java version your main thing. Either you can set the version you want as your "java home", making that version be universal for all other applications, or there should be a way inside TLauncher too. A quick google search should give you the details.

**Comment by mariumbacchus** [comment]
Gonna close this since it may have been fixed in the newest fabric release. Re-open if it still occurs.
